api/src/main/java/io/lumeer/engine/api/data/DataDocument.java

```diff
@@ -261,7 +261,13 @@ public boolean getBoolean(final String key, final boolean defaultValue) {
     *       if key nested path is not valid
     */
    public Date getDate(final String key) {
-      return (Date) getObject(key);
+      Long date = null;
+      try {
+         date = (Long) getObject(key);
+      } catch (ClassCastException e) {
+         return (Date) getObject(key);
+      }
+      return new Date(date);
```

Comment:
I am afraid this will fail if the value of `date` is `null`. I would suggest adding a check if the value of `date` is not `null` on the last line and only in that case pass the value to the constructor.